cmd/rendmp/src/blackbox.rs

```diff
@@ -0,0 +1,774 @@
+            f,
+            "rail0 status temperature: {}",
+            self.rail0_status_temperature
+        )?;
+        writeln!(
+            f,
+            "rail1 status temperature: {}",
+            self.rail1_status_temperature
+        )?;
+        writeln!(
+            f,
+            "rail2 status temperature: {}",
+            self.rail2_status_temperature
+        )?;
+        writeln!(f, "rail0 status input: {}", self.rail0_status_input)?;
+        writeln!(f, "rail1 status input: {}", self.rail1_status_input)?;
+        writeln!(f, "rail2 status input: {}", self.rail2_status_input)?;
+
+        writeln!(f)?;
+        writeln!(f, "     | RAIL 0  | RAIL 1  | RAIL 2")?;
+        writeln!(f, "-----|---------|---------|----------")?;
+        writeln!(
+            f,
+            "VIN  | {:<7} | {:<7} | :{}",
+            self.rail0_read_vin, self.rail1_read_vin, self.rail2_read_vin
+        )?;
+        writeln!(
+            f,
+            "VOUT | {:<7} | {:<7} | {}",
+            self.rail0_read_vout, self.rail1_read_vout, self.rail2_read_vout
+        )?;
+        writeln!(
+            f,
+            "IIN  | {:<7} | {:<7} | {}",
+            self.rail0_read_iin, self.rail1_read_iin, self.rail2_read_iin
+        )?;
+        writeln!(
+            f,
+            "IOUT | {:<7} | {:<7} | {}",
+            self.rail0_read_iout, self.rail1_read_iout, self.rail2_read_iout
+        )?;
+        writeln!(
+            f,
+            "TEMP | {:<7} | {:<7} | {}",
+            self.rail0_read_temperature,
+            self.rail1_read_temperature,
+            self.rail2_read_temperature
+        )?;
+
+        writeln!(
+            f,
+            "controller read temperature: {}",
+            self.controller_read_temperature
+        )?;
+        writeln!(f, "vmon: {}", self.vmon)?;
+        writeln!(f, "vinsen: {}", self.vinsen)?;
+
+        writeln!(f, " PHASE | TEMPERATURE | CURRENT ")?;
+        writeln!(f, "-------|-------------|----------")?;
+        for (i, (t, c)) in self
+            .phase_temperatures
+            .iter()
+            .zip(self.phase_currents.iter())
+            .enumerate()
+        {
+            writeln!(f, " {i:^5} | {t:^11} | {c}")?;
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, FromBytes)]
+struct ControllerFaultGen2p5(U32);
+impl Display for ControllerFaultGen2p5 {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        let v = self.0.get();
+        let faults = [
+            (23, "open pin"),
+            (12, "MCU fault"),
+            (9, "start-up fault"),
+            (7, "controller OT fault"),
+            (6, "PLL fault"),
+            (5, "PLL fault"),
+            (4, "oscillator fault"),
+            (2, "VCCS UV fault"),
+            (1, "VCC UV fault"),
+            (0, "VCC UV fault"),
+        ];
+        let faults: Vec<String> = faults
+            .iter()
+            .filter(|(i, _)| v & (1 << i) != 0)
+            .map(|(_, v)| v.to_string())
```

Comment:
Same nits as above - `filter_map()` and no `.to_string()`

Reply:
Not sure why this isn't using `format_faults`, so let's just do that instead.